store/sqlstore/user_store.go

```diff
@@ -44,25 +42,16 @@ type SqlUserStore struct {
 }
 
 var profilesInChannelCache *utils.Cache = utils.NewLru(PROFILES_IN_CHANNEL_CACHE_SIZE)
-var profileByIdsCache *utils.Cache = utils.NewLru(PROFILE_BY_IDS_CACHE_SIZE)
 
 func (us SqlUserStore) ClearCaches() {
 	profilesInChannelCache.Purge()
-	profileByIdsCache.Purge()
 
 	if us.metrics != nil {
 		us.metrics.IncrementMemCacheInvalidationCounter("Profiles in Channel - Purge")
-		us.metrics.IncrementMemCacheInvalidationCounter("Profile By Ids - Purge")
 	}
 }
 
-func (us SqlUserStore) InvalidatProfileCacheForUser(userId string) {
-	profileByIdsCache.Remove(userId)
-
-	if us.metrics != nil {
-		us.metrics.IncrementMemCacheInvalidationCounter("Profile By Ids - Remove")
-	}
-}
+func (us SqlUserStore) InvalidateProfileCacheForUser(userId string) {}
```

Comment:
Good catch on the typo but I think we should also leave the old function and deprecate it to keep backward compatibility due to [semver](https://semver.org/) requirements.

Reply:
You are right, I was eager to make the fix so I couldn't see the problem with the compatibility. I will change this @streamer45

Reply:
The store itself is in our internal API, is not exposed to be used for any third party, I'm 4/5 we don't need to be compatible with semver at this level?

Reply:
> The store itself is in our internal API, is not exposed to be used for any third party, I'm 4/5 we don't need to be compatible with semver at this level?

I tend to agree but at the same time we made those methods *public* from a developer perspective. I am fine with changing as long as we keep consistent with this and we well define the line between public API and internal API.

Reply:
I think it, for now, it is clearly an internal API, is not exposed outside mattermost in any way, and is not expected to be used directly from anybody outside the mattermost server development.